src/js/events/MetronomeActions.js

```diff
@@ -74,10 +65,21 @@ const MetronomeActions = {
               });
               resolve();
             } catch (error) {
-              handleError(error);
+              AppDispatcher.handleServerAction({
+                type: REQUEST_METRONOME_JOBS_ERROR,
+                data: error
+              });
+              reject();
             }
           },
-          error: handleError,
+          error(xhr) {
+            AppDispatcher.handleServerAction({
+              type: REQUEST_METRONOME_JOBS_ERROR,
+              data: xhr.message,
```

Comment:
I understand where you're coming from and agree that would be nice. I do think it's out of scope for this PR though and perhaps not necessary as we migrate to GraphQL. That's tackling technical debt that already existed. I just took the same code and put it in a different place so it was easier to read.